lib/iiif_print/data/fileset_helper_decorator.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+# OVERRIDE IIIF Print v1.0.0 to deal with a mismatch error
+
+IiifPrint::Data::FilesetHelper.module_eval do
+  def first_fileset
+    # TODO: This occured in UTK as well, so we should contribute this back to IIIF Print
+    # Instead of doing a FileSet.find, we do an ActiveFedora::Base.find to avoid a mismatch error
+    @work = ActiveFedora::Base.find(@work) if @work.is_a?(String)
+    return unless @work.is_a? FileSet
+    return @work if @work.is_a? FileSet
+
+    filesets = @work.members.select { |m| m.is_a? FileSet }
+    filesets.empty? ? nil : filesets[0]
```

Comment:
if we're doing an empty return if `@work` is anything _other_ than a file set....then we're returning `@work` if it _is_ a file set....how do we get to lines 13-14?

Reply:
Good catch! This was left over from the old Newspaper gem logic, as I'm looking at it now, I'm not sure why it would have worked since if `@work` comes in as something that is NOT a FileSet, then we'd get a type mismatch error like we're seeing.  If it came in as a string then we find it's FileSet and if it does come in as a FileSet then it just returns it.  So I'm not sure why you would call `#members` on a FileSet to look for more FileSets.  🤔 

https://github.com/scientist-softserv/iiif_print/blob/8fdf56e151b5adb7e6aab1cd29d39b74554ed48a/lib/iiif_print/data/fileset_helper.rb#L13-L22

Reply:
in the code you linked above, we would get to the code on your lines 13-14 in the instance that `@work` was a Work model.

however, in the code changes you have right now, you are returning from the method if `@work` is a Work model. so the code on 13-14 doesn't apply. if the intention is to get the file sets on a work, then I think line 10 needs to be removed.

in that case, we return the FileSet if that was passed in, or we find the FileSet if a Work was passed in.

Reply:
yes the intention is the get the FileSet, i have the IIIF Print PR you looked at that addresses this at a higher level so actually i'm going to close this one and update IIIF Print instead, thanks for taking a look!

Reply:
Actually, going to keep this PR opened and push another commit to update IIIF Print, might as well keep it in one instead of splitting it into two